frontends/election-manager/src/components/export_ballot_pdfs_button.tsx

```diff
@@ -0,0 +1,375 @@
+                    setIsAbsentee={setIsAbsentee}
+                    absenteeFirst={false}
+                  />
+                </p>
+              </CenteredOptions>
+              <p>
+                A zip archive will automatically be saved to the default
+                location on the mounted USB drive. Optionally, you may pick a
+                custom export location.
+              </p>
+            </Prose>
+          );
+          actions = (
+            <React.Fragment>
+              <Button primary onPress={() => saveFileCallback(false)}>
+                Export
+              </Button>
+              <LinkButton onPress={closeModal}>Cancel</LinkButton>
+              <Button onPress={() => saveFileCallback(true)}>Custom</Button>
+            </React.Fragment>
+          );
+          break;
+        }
+
+        default:
+          throwIllegalValue(usbDriveStatus);
+      }
+      break;
+
+    case ModalState.GENERATE_FILES: {
+      mainContent = (
+        <React.Fragment>
+          <Prose textCenter>
+            <h1>
+              <strong>Generating Ballot PDFs…</strong>
+            </h1>
+          </Prose>
+          <PrintableArea>
+            <HandMarkedPaperBallot
+              ballotStyleId={ballotStyle.ballotStyleId}
+              election={election}
+              electionHash={electionHash}
+              ballotMode={ballotMode}
+              isAbsentee={isAbsentee}
+              precinctId={ballotStyle.precinctId}
+              onRendered={onRendered}
+              locales={defaultBallotLocales}
+            />
+          </PrintableArea>
+        </React.Fragment>
+      );
+      break;
+    }
+
+    case ModalState.END_EXPORT: {
+      mainContent = (
+        <Prose textCenter>
+          <h1>
+            <strong>Finishing Export…</strong>
+          </h1>
+        </Prose>
+      );
+      break;
+    }
+
+    case ModalState.DONE: {
+      mainContent = (
+        <Prose>
+          <h1>Export Complete</h1>
+          <p>You may now eject the USB drive.</p>
+        </Prose>
+      );
+      if (usbDriveStatus !== UsbDriveStatus.recentlyEjected) {
+        actions = (
+          <React.Fragment>
+            <UsbControllerButton
+              primary
+              small={false}
+              usbDriveEject={() => usbDriveEject(userRole)}
+              usbDriveStatus={usbDriveStatus}
+            />
+            <LinkButton onPress={closeModal}>Close</LinkButton>
+          </React.Fragment>
+        );
+      } else {
+        actions = <LinkButton onPress={closeModal}>Close</LinkButton>;
+      }
+      break;
+    }
+
+    case ModalState.ERROR: {
+      mainContent = (
+        <Prose>
+          <h1>Export Failed</h1>
+          <p>An error occurred: {modalError && modalError.message}.</p>
+        </Prose>
+      );
+      actions = <LinkButton onPress={closeModal}>Close</LinkButton>;
+      break;
+    }
+
+    default:
+      // nothing to do
```

Comment:
To make sure that you don't forget to cover any states in the `switch` `case`, adding a `throwIllegalValue` call here will make TS ensure that you've covered everything:

https://github.com/votingworks/vxsuite/blob/3df0503a5e13fcca49f3b4036bbb92787025393b/frontends/election-manager/src/components/smartcard_modal/status_message.tsx#L45-L48